Android/src/org/droidplanner/android/maps/providers/baidu_map/BaiduMapFragment.java

```diff
@@ -0,0 +1,928 @@
+            MyLocationData locData = new MyLocationData.Builder()
+                    .accuracy(bdloc.getRadius())
+                    .direction(100).latitude(bdloc.getLatitude())
+                    .longitude(bdloc.getLongitude()).build();
+
+            // convert BDLocation to android.Location
+            Location aLoc = new Location("BaiduMap");
+            //aLoc.setBearing(); No bearing in baidu map
+            aLoc.setLongitude(bdloc.getLongitude());
+            aLoc.setLatitude(bdloc.getLatitude());
+            mLocationListener.onLocationChanged(aLoc);
+        }
+    }
+
+    private void updateBDMapStatus(BDLocation location) {
+        final BaiduMap map = getBaiduMap();
+        final BaiduMapPrefFragment provider = (BaiduMapPrefFragment)(getProvider().getMapProviderPreferences());
+        final Activity activity = getActivity();
+        final Context context = activity.getApplicationContext();
+
+        int mapType =provider.getMapType(context);
+        map.setMapType(mapType);
+
+        MyLocationData locData = new MyLocationData.Builder()
+                .accuracy(location.getRadius())
+                .direction(100).latitude(location.getLatitude())
+                .longitude(location.getLongitude()).build();
+        map.setMyLocationData(locData);
+
+        LatLng ll = new LatLng(location.getLatitude(), location.getLongitude());
+        MapStatusUpdate u = MapStatusUpdateFactory.newLatLng(ll);
+        map.animateMapStatus(u);
+    }
+
+    public class MyBDLocationListenner implements BDLocationListener {
+		@Override
+        public void onReceiveLocation(BDLocation location) {
+			final BaiduMap map = getBaiduMap();
+			final Activity activity = getActivity();
+			if (map == null || location == null || activity == null)
+				return;
+
+            updateBDMapStatus(location);
+            notifyToLocatorActivity(location);
+        }
+
+        public void onReceivePoi(BDLocation poiLocation) {
+        }
+    }
+
+
+    // FIXME: Ugly, Duplicated lat long titude convert functions, have to be templated
+    private LatLng CoordToLatLang(LatLong coord) {
+        return new LatLng(coord.getLatitude(), coord.getLongitude());
+    }
+
+    private LatLong LatLngToCoord(LatLng point) {
+        return new LatLong((float)point.latitude, (float) point.longitude);
+    }
+
+    private LatLong LocationToCoord(BDLocation location) {
+        return new LatLong((float) location.getLatitude(), (float) location.getLongitude());
+    }
+
+    private int scaleDpToPixels(double value, Resources res) {
+        final float scale = res.getDisplayMetrics().density;
+        return (int) Math.round(value * scale);
+    }
+
+    private Drone getDroneApi() { return mDpApp.getDrone(); }
+
+    @Override
+    public void onAttach(Activity activity) {
+        super.onAttach(activity);
+
+        // Initialize Baidu Map library
+        com.baidu.mapapi.SDKInitializer.initialize(getActivity().getApplicationContext());
```

Comment:
@droneboost use the `activity` parameter instead of calling the `getActivity()` method.